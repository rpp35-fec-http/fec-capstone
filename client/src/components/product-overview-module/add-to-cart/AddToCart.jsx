import React, { useState } from 'react';
import './AddToCart.scss';
import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';

const AddToCart = ({ currentStyle }) => {
  const [currentSize, setCurrentSize] = useState('');
  const [purchaseQuantity, setPurchaseQuantity] = useState(0);
  const [quantity, setQuantity] = useState([]);

  //handles turning skus into array so that they can be mapped over
  const skus = Object.keys(currentStyle.skus).map((key) => {
    return currentStyle.skus[key];
  });

  //handles when size is selected to get correct quantity and size options
  const handleChange = (e) => {
    setCurrentSize(e.target.value);
    let quantityArr = [];
    let currentQuantity;
    for (let i = 0; i < skus.length; i++) {
      if (skus[i].size === e.target.value) {
        if (skus[i].size >= 15) {
          currentQuantity = 15;
        } else {
          currentQuantity = skus[i].quantity;
        }
      }
    }
    for (let i = 1; i < currentQuantity; i++) {
      quantityArr.push(i);
    }
    setQuantity(quantityArr);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
  };

  //typo in data - has two options with XL size
  if (skus.length) {
    skus[skus.length - 1].size = 'XXL';
  }
  ////////////////////////////////////////////

  //handles changing state when choosing quantity
  const handleSelectQuantity = (e) => {
    setPurchaseQuantity(e.target.value);
  };
  return (
    <form className='form' onSubmit={handleSubmit}>
      {skus.length ? (
        <>
          <select className='form-select-size' required onChange={handleChange}>
            <option value=''>Select Size</option>
            {skus.map((itemSize, index) => (
              <option key={index} label={itemSize.size}>
                {itemSize.size}
              </option>
            ))}
          </select>
          <select className='form-quantity' required>
            {quantity.length ? <option>1</option> : <option disabled>-</option>}
            {quantity.map((number) => (
              <option key={number} label={number}>
                {number}
              </option>
            ))}
          </select>
          <Button className='form-button' variant='contained' type='submit'>
            Add To Cart
          </Button>
        </>
      ) : (
        <select disabled>
          <option>OUT OF STOCK</option>
        </select>
      )}
    </form>
  );
};

export default AddToCart;
