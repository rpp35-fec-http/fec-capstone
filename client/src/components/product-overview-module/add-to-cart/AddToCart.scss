@use '../../App.scss' as *;

.form {
  display: flex;
  justify-content: space-evenly;
  .form-select-size {
    @include box(12rem, 75px);
  }
  .form-quantity {
    @include box(12rem, 75px);
  }
  .form-button {
    @include box(8rem, 75px);
  }
  option {
    font-size: 20px;
  }
}
