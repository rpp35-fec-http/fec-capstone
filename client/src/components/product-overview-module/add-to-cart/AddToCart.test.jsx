import React from 'react';
import renderer from 'react-test-renderer';
import AddToCart from './AddToCart.jsx';

it('renders correctly', () => {
  let currentStyle = {};
  let skus = {
    234: { quantity: 8, size: 'S' },
    23444: { quantity: 2, size: 'M' },
  };
  currentStyle.skus = skus;
  const tree = renderer
    .create(<AddToCart currentStyle={currentStyle} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
