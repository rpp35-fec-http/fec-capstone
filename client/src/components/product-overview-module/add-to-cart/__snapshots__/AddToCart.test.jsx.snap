// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <select
    className="form-select-size"
    onChange={[Function]}
    required={true}
  >
    <option
      value=""
    >
      Select Size
    </option>
    <option
      label="S"
    >
      S
    </option>
    <option
      label="XXL"
    >
      XXL
    </option>
  </select>
  <select
    className="form-quantity"
    required={true}
  >
    <option
      disabled={true}
    >
      -
    </option>
  </select>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root form-button css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    Add To Cart
  </button>
</form>
`;
